lib/pip.js

```diff
@@ -286,7 +286,7 @@ function generateRequirementsFile(source, target, options) {
       return false;
     }
     // If we have requirements.txt options (prefixed with --) keep them
-    if (req.length > 1 && req[0] == '-' && req[1] == '-') {
+    if (req.length > 1 && req.startsWith('--')) {
```

Comment:
you can also remove the length check. `''.startsWith('--') === false`

Reply:
Done, and done to the above check as well